src/visit/traversal.rs

```diff
@@ -344,6 +344,18 @@ where
         topo
     }
 
+    /// Create a new `Topo` with initial nodes
+    pub fn with_initials<G, I>(graph: G, i: I) -> Self
+    where
+        G: Visitable<NodeId = N, Map = VM>,
+        I: IntoIterator<Item = N>,
+    {
+        Topo {
+            tovisit: i.into_iter().collect(),
+            ordered: graph.visit_map(),
+        }
```

Comment:
If the user passes a node with incoming edges, this would then yield duplicates.

I say mark the items in `i` in the `visit_map`. Then we'd process them instead as if the incoming edges didn't exist.

Reply:
The items in `i` could be children of others nodes, then we may miss them if we just mark them in the `visit_map`. It's more reasonable to filter the nodes with incoming edges I think.